### PR TITLE
DAC6-968: Implement Upscan changes

### PR DESCRIPTION
**Pull request by ers2015112**
commit e701951f1f826ebef6f528231a369f4e994742dd
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 20 17:32:54 2021 +0100

    DAC6-968:  Until the build jobs changes are implemented set feature flag to false

commit 7541348e086ead9af24398b3c9158bb80b832a94
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 20 16:16:06 2021 +0100

    DAC6-968: update SubmissionControllerSpec

commit d82710f4e90eb5c5a852b3c263193b1087da1fb3
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Fri Aug 20 15:11:37 2021 +0100

    DAC6-968: Update tess with NoEnrolmentIdentifier

commit cf68e9a37ff3a92be7ad5edc2520274db3863f2b
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 20 14:33:25 2021 +0100

    DAC6-968: fetch file validation URL based on feature switch

commit 7541488b5b1ac04972a6867e7d9931ad8c6a0031
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 20 12:04:20 2021 +0100

    DAC6-968: fetch file validation URL based on feature switch

commit 3550661a0b90be217551cdb47697d1de3a08573b
Merge: a737b555 b2dd4220
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Thu Aug 19 16:09:47 2021 +0100

    Merge remote-tracking branch 'origin/DAC6-968' into DAC6-968

commit a737b555c58546cbd6ce8e349d2991ec409b2ad2
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Thu Aug 19 16:06:35 2021 +0100

    DAC6-968: fetch file validation URL based on feature switch

commit b2dd4220702f225c566c5d3aefa94b17a4a0e6fc
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Thu Aug 19 12:19:32 2021 +0100

    DAC6-968: Complete fileValiadation controller tests

commit 1380b1fab77b74cb842e1c4f528e2b66128e49d6
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Thu Aug 19 11:32:16 2021 +0100

    DAC6-968: Update test

commit 5058efc66a60ef89c0f633ce8d329704b4ac1784
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Thu Aug 19 09:55:08 2021 +0100

    DAC6-968: Update test

commit c493ad1a614a917b4f41ad2cf66b8a604b4e3c78
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Wed Aug 18 17:18:30 2021 +0100

    DAC6-968: Check for cbc new feature flag before Submission

commit 8b1a387bdae7504b11a6e8601d68da04ff7add67
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 17 16:07:41 2021 +0100

    DAC6-968: Call upscan URL's

commit f8560c3cdd38b21e5b2486d51f9a95eb7a71f50c
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 17 15:12:55 2021 +0100

    DAC6-968: Remove println and update back link

commit fa8f68fd1e71cc05174d44b5a59e1529853daca3
Merge: 49b22c2e 7968c62d
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 17 15:01:28 2021 +0100

    Merge remote-tracking branch 'origin/DAC6-968' into DAC6-968

    # Conflicts:
    #	app/uk/gov/hmrc/cbcrfrontend/services/FileService.scala
    #	app/uk/gov/hmrc/cbcrfrontend/services/FileValidationService.scala
    #	app/uk/gov/hmrc/cbcrfrontend/views/Views.scala
    #	app/uk/gov/hmrc/cbcrfrontend/views/upscan/uploadError.scala.html
    #	app/uk/gov/hmrc/cbcrfrontend/views/upscan/uploadForm.scala.html
    #	app/uk/gov/hmrc/cbcrfrontend/views/upscan/uploadProgress.scala.html
    #	app/uk/gov/hmrc/cbcrfrontend/views/upscan/uploadResult.scala.html
    #	conf/app.routes

commit 7968c62d2aaf604ee0e362b74316bc83ffbfb9f7
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Tue Aug 17 14:55:46 2021 +0100

    DAC6-968: rebase to master

commit 49b22c2e8f8ab52567fd21d92983c9e5a822e633
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 17 13:33:57 2021 +0100

    DAC6-968: Redirect to upscan controller when feature enhancement flag is true

commit 98c54dc65aabb52ed8715aea472c31f5ba27678b
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Tue Aug 17 13:30:16 2021 +0100

    DAC6-968: WIP

commit 1fb594f85b5c86a956f85da098aa46bebaacba4b
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 18:01:58 2021 +0100

    DAC6-968: Update FileValidationController to use FileValidateService

commit 31197b969ee843f5bf5e934ec444fec92778fb5d
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 16:55:24 2021 +0100

    DAC6-968: refactor common component

commit e578f02e0f39fb33e33ccefa612a0b109e635fb7
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 16 16:43:52 2021 +0100

    DAC6-968: WIP

commit d44e5fe0edc07f95b8b1bd6e2062cd530f15b9f7
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 16 15:33:35 2021 +0100

    DAC6-968: Add fileValidation service tests

commit e70b905cd2224d0430780be48a27d1fb506a524d
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 16 14:54:47 2021 +0100

    DAC6-968: WIP

commit da3fd8e379fdc0179103437624290da7da6d927f
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 11:12:34 2021 +0100

    DAC6-968: Update AuditServiceSpec to test errorState

commit 911ba07260221c3e6be929af477a723ccd682908
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 10:06:52 2021 +0100

    DAC6-968: Update AuditServiceSpec to test errorState

commit b1a48e6e9a980d7e0bede65a4a1cc5b85132dbe2
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 10:03:54 2021 +0100

    DAC6-968: Add AuditServiceSpec to test the Audit Service

commit 2ae03fab8c734bdc42edafaaeafd596dbfaad4e1
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 09:04:55 2021 +0100

    DAC6-968: Add AuditServiceSpec to test the Audit Service

commit d587abdfb1c8011300bfebd3f9cb0037a9f8c9d6
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 13 18:35:31 2021 +0100

    DAC6-968: Add test to validate file

commit cf330134d6bbb7bdec22e383671bbe75a7425028
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Fri Aug 13 14:32:36 2021 +0100

    DAC6-968: WIP

commit 02c3cd803d5dca6a39a698d15a5a99e8a1b484a2
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Fri Aug 13 14:05:04 2021 +0100

    DAC6-968: WIP

commit 8dc7c606fced2e8945ab63d5d2c19ed36e104a68
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 13 09:32:37 2021 +0100

    DAC6-968: Update SpecBase

commit ce1fefc8e35079fe3264ee56d5a2117d58873325
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 13 08:02:07 2021 +0100

    DAC6-968: Add audit service and update validation service

commit b62ce16ef3d226d90e907a503e2c09f4394d2211
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Thu Aug 12 16:51:27 2021 +0100

    DAC6-968: WIP

commit e30e71fb5fe48deb42e9a3eff36171b168057d43
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Thu Aug 12 15:38:12 2021 +0100

    DAC6-968: WIP

commit 00fbdeccce374e4e8435dd7de22061021ec92c4d
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Wed Aug 11 16:45:47 2021 +0100

    DAC6-968: Call the correct upscan endpoint

commit 5eabe6b16a82633faddaae545be3585eddd13f2e
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 16:27:59 2021 +0100

    DAC6-968: Update call back URL's

commit e3422eba679706bc173a99b36e44231b41c6ce79
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 14:04:56 2021 +0100

    DAC6-968: Update uploadForm UI to submit to upscan

commit 3b86316c3898aa8ec0ea1bf46b32873c416ceb97
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 11:22:00 2021 +0100

    merge with master

commit 33ecd28680668aa0ba50c9c690a54170ef207c79
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 10:50:10 2021 +0100

    DAC6-968: Update uploadForm UI to submit to upscan

commit d69c8d6264a5c14dbe19382990d27057548f9265
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 9 16:04:20 2021 +0100

    DAC6-968: WIP

commit 539685bd5301cf8e7be874889d2185465d3a0f56
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 9 15:13:15 2021 +0100

    Add controller test for upscan

commit f3fd55bbbd29fb2bb4450848f433b596d18f48d2
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 9 11:19:28 2021 +0100

    DAC6-968: Add uploadIn-progress UI

commit 563ba251e06bfd421c3ba19dc8de59d013d7e466
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 6 16:36:16 2021 +0100

    DAC6-968: Add upload UI

commit b3088acd48e9d48f370537b5f4927eb923e18d45
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Fri Aug 6 16:14:08 2021 +0100

    DAC6-968-New: Wip

commit 763cc7cf8108feceb74de022cb9cdc9184fede99
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 17 09:45:45 2021 +0100

    DAC6-968: Update FileValidationController to use FileValidateService

commit 9b74ccb27be3210fe586fa80fb8c500a1b52b7b7
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 18:01:58 2021 +0100

    DAC6-968: Update FileValidationController to use FileValidateService

commit 633d546fc3e18a3e4530a5245f6f15a1298bc76c
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 16:55:24 2021 +0100

    DAC6-968: refactor common component

commit 2fd90cb2ad42e7ea4e295a062c7031ec55125fee
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 16 16:43:52 2021 +0100

    DAC6-968: WIP

commit d9787fd65b49d97d47797fd0586f1b55f81e18d0
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 16 15:33:35 2021 +0100

    DAC6-968: Add fileValidation service tests

commit 9cb32fff98f393e90d097d7335ea9bf5c20e8ed7
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 16 14:54:47 2021 +0100

    DAC6-968: WIP

commit 9c49bfae78c341725d10de28c4c8d52a4c11506e
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 11:12:34 2021 +0100

    DAC6-968: Update AuditServiceSpec to test errorState

commit fe6176a93898a5e8ac6f9f36290e8e910594ca02
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 10:06:52 2021 +0100

    DAC6-968: Update AuditServiceSpec to test errorState

commit 2f7aede90cb5eab0d1fd6818916cc6f42570f0ab
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 10:03:54 2021 +0100

    DAC6-968: Add AuditServiceSpec to test the Audit Service

commit b86fdac5cf36e60a4158fef1d6badbf73afb90ac
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 16 09:04:55 2021 +0100

    DAC6-968: Add AuditServiceSpec to test the Audit Service

commit 1e1b9ead59008e1bfc2ba63a82d64ed5e0fd8b94
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 13 18:35:31 2021 +0100

    DAC6-968: Add test to validate file

commit ca95bec67c5fe803a9d601639a768dc585db2ca6
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Fri Aug 13 14:32:36 2021 +0100

    DAC6-968: WIP

commit 51ab999b711d9395bd24427c0e38feecc745b369
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Fri Aug 13 14:05:04 2021 +0100

    DAC6-968: WIP

commit b92aa43764b0b8dfd0640a2e119de33defccf386
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 13 09:32:37 2021 +0100

    DAC6-968: Update SpecBase

commit 254c50b32500a5ed51c032afed80fde223a999bd
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 13 08:02:07 2021 +0100

    DAC6-968: Add audit service and update validation service

commit 8da596d7ce6f940a95f12a4e44524a5daaba9686
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Thu Aug 12 16:51:27 2021 +0100

    DAC6-968: WIP

commit e9dec2bad16f9703926776dc2331e386dfd8c323
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Thu Aug 12 15:38:12 2021 +0100

    DAC6-968: WIP

commit 01529e8e15bb6e8a27bf48a52c5e42d4272fcb59
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Wed Aug 11 16:45:47 2021 +0100

    DAC6-968: Call the correct upscan endpoint

commit 6ed5800737949d6776312d02c1da9b2e1257051a
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 16:27:59 2021 +0100

    DAC6-968: Update call back URL's

commit 91804ef10a4ac65ac1aa57574876e3a8963db9a0
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 14:04:56 2021 +0100

    DAC6-968: Update uploadForm UI to submit to upscan

commit 4daf2a76a54f97ce2efdf479c6f7d7da1cd5540f
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 11:22:00 2021 +0100

    merge with master

commit 9e89dfd761844fe61cba5b114a7ae9fd592f1525
Merge: 89397149 c0aa78d1
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 11:05:18 2021 +0100

    Merge remote-tracking branch 'origin/master' into DAC6-968

    # Conflicts:
    #	app/uk/gov/hmrc/cbcrfrontend/views/Views.scala
    #	conf/app.routes

commit 89397149a33e0fe8be7b622e6e90aae51344281c
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Tue Aug 10 10:50:10 2021 +0100

    DAC6-968: Update uploadForm UI to submit to upscan

commit cab22b461f3f1d096a879e0c238db52bf457b940
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Mon Aug 9 16:04:20 2021 +0100

    DAC6-968: WIP

commit bc71e66ef424aab2266bb7db6f281b0d5cb1cb60
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 9 15:13:15 2021 +0100

    Add controller test for upscan

commit 1b1c8ca7b365b26093fe0d04ea58e7076bcc0034
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Mon Aug 9 11:19:28 2021 +0100

    DAC6-968: Add uploadIn-progress UI

commit f873c02854b364eeebf4ba44f4047833568200b3
Author: ashwinirudrappa <11343826+ashwinirudrappa@users.noreply.github.com>
Date:   Fri Aug 6 16:36:16 2021 +0100

    DAC6-968: Add upload UI

commit 820c71905b9e3f86570b754146488b091f8ded3b
Author: ers2015112 <20478032+ers2015112@users.noreply.github.com>
Date:   Fri Aug 6 16:14:08 2021 +0100

    DAC6-968-New: Wip

# JIRA Story or Github Number - descriptive title

**Bug fix / New feature** (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala [hunk 1]
```diff
@@ -219,7 +219,7 @@ class FileUploadControllerSpec
 
   val testFile: File = new File("test/resources/cbcr-valid.xml")
   val tempFile: File = File.createTempFile("test", ".xml")
-  val validFile = java.nio.file.Files.copy(testFile.toPath, tempFile.toPath, REPLACE_EXISTING).toFile
+  val validFile: File = java.nio.file.Files.copy(testFile.toPath, tempFile.toPath, REPLACE_EXISTING).toFile
```
Comment (iservport) [inline test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala:222]: TDD: see line 234 below, there is no URL to "GET /upload-report".

File: app/uk/gov/hmrc/cbcrfrontend/services/FileValidationService.scala [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      uploadId       <- getUploadId()
+      uploadSessions <- upscanConnector.getUploadDetails(uploadId)
+      uploadDetails = getUploadDetails(uploadSessions)
+    } yield (uploadDetails, uploadId)
+
+    for {
+      file_meta <- right[(UploadedSuccessfully, UploadId)](fileDtls)
+      file      <- fileService.getFile(file_meta._2, file_meta._1.downloadUrl)
+      _         <- right(cache.save[UploadedSuccessfully](file_meta._1))
+      _         <- EitherT.cond[Future](file_meta._1.name.toLowerCase endsWith ".xml", (), InvalidFileType(file_meta._1.name))
```
[the remark this answers is not in the document]
Comment (ashwinirudrappa) [reply app/uk/gov/hmrc/cbcrfrontend/services/FileValidationService.scala:67]: @iservport it is for validation, is it is anything other then xml it throws error

File: test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala:222]: @iservport Its FileUpload not upscan, this code will be removed

**Review by iservport — APPROVED** [review]
LGTM

**Pull request merged**